pkg/agent/addon_status_helper.go

```diff
@@ -0,0 +1,59 @@
+package agent
+
+import (
+	"fmt"
+
+	"github.com/stolostron/hypershift-addon-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+)
+
+func containsHypershiftAddonDeployment(deployment appsv1.Deployment) bool {
+	if len(deployment.Name) == 0 || len(deployment.Namespace) == 0 {
+		return false
+	}
+
+	if deployment.Namespace != util.HypershiftOperatorNamespace {
+		return false
+	}
+
+	return deployment.Name == util.HypershiftOperatorExternalDNSName ||
+		deployment.Name == util.HypershiftOperatorName
+}
+
+func generateNoAvailableDegradedReason(deploymentName string) string {
+	switch deploymentName {
+	case util.HypershiftOperatorName:
+		return operatorDegradedReason
+	case util.HypershiftOperatorExternalDNSName:
+		return externalDNSDegradedReason
+	}
+
+	return fmt.Sprintf("No%sAvailable", deploymentName)
+}
+
+func checkDeployment(deployment appsv1.Deployment, notFoundError error) metav1.Condition {
+	addonCondtion := metav1.Condition{
+		Type:   addonv1alpha1.ManagedClusterAddOnConditionDegraded,
+		Status: metav1.ConditionTrue,
+		Reason: generateNoAvailableDegradedReason(deployment.Name),
+	}
+
+	if notFoundError != nil {
+		addonCondtion.Message = fmt.Sprintf("The %s deployment does not exist", deployment.Name)
+		return addonCondtion
+	} else if !deployment.GetDeletionTimestamp().IsZero() {
+		addonCondtion.Message = fmt.Sprintf("The %s deployment is being deleted", deployment.Name)
+		return addonCondtion
+	} else if deployment.Status.AvailableReplicas == 0 {
```

Comment:
Should this > 0? Or  == deployment.Spec.replicas??

Reply:
Sounds good. I will double check what others do to follow convention.

Reply:
Done as suggested and introduced unique reasons for different failures.